common/src/stack/report-system/command/report/system/fixtures/disk_partition_check.py

```diff
@@ -0,0 +1,191 @@
+
+	lsblk_data = stdout.readlines()
+
+	partition_attributes = namedtuple("partition", attributes)
+
+	# For each line in the lsblk output, make it into a tuple
+	# Then each partition into a dictionary keyed by it's disk
+	if lsblk_data:
+		for line in lsblk_data:
+			part_values = partition_attributes(*line.rstrip().split(' '))
+
+			# Convert bytes to megabytes if the size column was specified
+			if 'size' in attributes:
+				part_values = part_values._replace(size = int(part_values.size) / math.pow(2, 20))
+
+			# Get the name of the disk by taking the partition name without it's
+			# partition number
+			disk = ''.join(filter(lambda x: not x.isdigit(), part_values.name))
+
+			disks.setdefault(disk, {})[part_values.name] = part_values
+	else:
+		return {}
+
+	return disks
+
+
+def check_partition_size(partitions, curr_partition, expect_size, disk):
+	part_size = 0
+
+	if expect_size == '*':
+		return False
+
+	elif int(expect_size) == 0:
+		for partition, attributes in partitions.items():
+			if partition != disk and partition != curr_partition.name:
+				part_size += int(attributes.size)
+
+		remain_size = int(partitions[disk].size) - int(part_size)
+
+		if math.isclose(int(remain_size), int(curr_partition.size), abs_tol=100):
+			return True
+		else:
+			return False
+
+	elif not math.isclose( int(expect_size), int(curr_partition.size), abs_tol=100):
+		return False
+
+	else:
+		return True
+
+
+@pytest.fixture
+def verify_disk():
+	def partition_match_expected(expected_partitions, hostname, check_all_disks=False):
+
+		lsblk_col = ['name', 'mountpoint', 'label', 'fstype', 'size']
+
+		# Get the actual partition info from the host
+		part_data = get_partition_data(hostname, lsblk_col)
+
+		# If we could not get it, return empty dict
+		if not part_data:
+			pytest.skip(f'Could not get partition data from host {hostname}')
+
+		matched_disks = {}
+
+		# Go through each disk
+		for disk, partitions in part_data.items():
+			matched_partitions = {}
+
+			# Go through each partition that we expect for the disk
+			# and see if it is actually there
+			if check_all_disks:
+				for partition, attributes in partitions.items():
+
+					if partition == disk:
+						continue
+
+					try:
+						check_part = expected_partitions[disk][partition]
+
+						attribute_not_match = {
+							field: getattr(attributes, field)
+							for field in check_part._fields if getattr(attributes, field)
+							!= getattr(check_part, field) and  field != 'size'
+						}
+
+						if 'size' in check_part._fields:
+							part_match = check_partition_size(partitions, attributes, check_part.size, disk)
+							if not part_match:
+								attribute_not_match['size'] = attributes.size
+
+						if attribute_not_match:
+							matched_partitions[partition] = attribute_not_match
+
+						else:
+							continue
+
+					except KeyError:
+						matched_partitions[partition] = {}
+
+				for check_disk in expected_partitions.keys():
+					try:
```

Comment:
This could be simplified by using `in` to test if the `check_disk` key is in the `part_data` dictionary:
```
# Iterating a dictionary yields its keys, so .keys() is also optional
for check_disk in expected_partitions:
    if check_disk not in part_data:
        matched_disks[check_disk] = {}
```